crates/pathfinder/src/state/merkle_node.rs

```diff
@@ -223,7 +223,7 @@ impl EdgeNode {
 
         let child = match self.child.borrow().hash() {
             Some(hash) => hash,
-            None => return,
+            None => todo!("this probably shouldn't be called with dirty children"),
```

Comment:
One alternative is to automatically resolve child hashes recursively, but not a good idea since it would hide a bug in our logic.

Reply:
I think this could be an assertion because execution doesn't seem to be every reaching this.

Reply:
Yes, it seems it is mostly called after committing the subtree.

Reply:
I'll leave this as unreachable, and I tried to lower the visibility of the two methods.